integration-test/src/test/resources/test_python_api.py

```diff
@@ -0,0 +1,173 @@
+    base64_pickled_job = base64.b64encode(pickled_job).decode('utf-8')
+    base64_pickled_job_json = json.dumps({'job': base64_pickled_job})
+    request_url = livy_end_point + "/sessions/" + str(session_id) + "/submit-job"
+    header = {'Content-Type': 'application/json', 'X-Requested-By': 'livy'}
+    response = requests.request('POST', request_url, headers=header, data=base64_pickled_job_json)
+
+    assert response.status_code == 201
+    job_id = response.json()['id']
+
+    poll_time = 0.1
+    max_poll_time = 10
+    poll_response = None
+    while (poll_response is None or poll_response.json()['state'] == 'STARTED') and poll_time < \
+            max_poll_time:
+        time.sleep(poll_time)
+        poll_request_uri = livy_end_point + "/sessions/" + str(session_id) + \
+                           "/jobs/" + str(job_id)
+        poll_header = {'X-Requested-By': 'livy'}
+        poll_response = requests.request('GET', poll_request_uri, headers=poll_header)
+        poll_time *= 2
+
+    assert poll_response.status_code == 200
+    assert poll_response.json()['id'] == job_id
+    result = poll_response.json()['result']
+    b64_decoded = base64.b64decode(result)
+    b64_decoded_decoded = base64.b64decode(b64_decoded)
+    deserialized_object = cloudpickle.loads(b64_decoded_decoded)
+    assert deserialized_object == expected_result
+
+
+def test_create_session():
+    global session_id
+
+    request_url = livy_end_point + "/sessions"
+    uri = urlparse(request_url)
+    header = {'Content-Type': 'application/json', 'X-Requested-By': 'livy'}
+    json_data = json.dumps({'kind': 'pyspark', 'conf': {'livy.uri': uri.geturl()}})
+    response = requests.request('POST', request_url, headers=header, data=json_data)
+
+    assert response.status_code == 201
+    session_id = response.json()['id']
+
+
+def test_spark_job():
+    time.sleep(40) # Internal Server Error - Session state in Starting - Need to figure this out
+
+    def simple_spark_job(context):
+        elements = [10, 20, 30]
+        sc = context.sc
+        return sc.parallelize(elements, 2).count()
+
+    process_job(simple_spark_job, str(3))
+
+
+def test_add_file():
+    add_file_name = pathlib2.Path(add_file_url).name
+    json_data = json.dumps({'uri': add_file_url})
+    request_url = livy_end_point + "/sessions/" + str(session_id) + "/add-file"
+    header = {'Content-Type': 'application/json', 'X-Requested-By': 'livy'}
+    response = requests.request('POST', request_url, headers=header, data=json_data)
+
+    assert response.status_code == 200
+
+    def add_file_job(context):
+        from pyspark import SparkFiles
+        with open(SparkFiles.get(add_file_name)) as testFile:
+            file_val = testFile.readline()
+        return file_val
+
+    process_job(add_file_job, "hello from addfile")
+
+
+def test_add_pyfile():
+    add_pyfile_name_with_ext = pathlib2.Path(add_pyfile_url).name
+    add_pyfile_name_list = add_pyfile_name_with_ext.rsplit('.', 1)
```

Comment:
You can avoid this temp variable by just adding `[0]` at the end of the statement; same later on.